src/test/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/ReportControllerTest.java

```diff
@@ -0,0 +1,136 @@
+import lombok.val;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import uk.gov.hmcts.reform.sandl.snlevents.model.report.ListedHearingRequestReportResult;
+import uk.gov.hmcts.reform.sandl.snlevents.model.report.UnlistedHearingRequestsReportResult;
+import uk.gov.hmcts.reform.sandl.snlevents.service.ReportService;
+
+import java.time.Duration;
+import java.time.LocalDate;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(ReportController.class)
+public class ReportControllerTest {
+    public static final String URL = "/report";
+
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private ReportService reportService;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Test
+    public void getUnlistedHearingRequests_returnsResultsFromService() throws Exception {
+        val results = createUnlistedHearingRequestReportResults();
+
+        when(reportService.reportUnlistedHearingRequests())
+            .thenReturn(results);
+
+        getResponseAndExpectOk(URL + "/unlisted-hearing-requests");
+    }
+
+    private List<UnlistedHearingRequestsReportResult> createUnlistedHearingRequestReportResults() {
+        return new ArrayList<>(Arrays.asList(new UnlistedHearingRequestsReportResult() {
+            @Override
+            public String getTitle() {
+                return null;
+            }
+
+            @Override
+            public int getHearings() {
+                return 0;
+            }
+
+            @Override
+            public int getMinutes() {
+                return 0;
+            }
+        }));
+    }
+
+    @Test
+    public void getListedHearingRequests_returnsResultsFromService() throws Exception {
+        val startDate = LocalDate.of(1,2,3);
+        val endDate = LocalDate.of(4,5,6);
+
+        val startDateString = "03-02-0001";
+        val endDateString = "06-05-0004";
+
+        val results = createListedHearingRequestReportResults();
+
+        when(reportService.reportListedHearingRequests(eq(startDate), eq(endDate)))
+            .thenReturn(results);
+
+        getResponseAndExpectOk(
+            URL + "/listed-hearing-requests?startDate=" + startDateString + "&endDate=" + endDateString
+        );
+    }
+
+    private void getResponseAndExpectOk(String url) throws Exception {
```

Comment:
Probably a good candidate for creating some kind of "TestHelper"

Reply:
I've extracted most common operations to OurMockMvc and yes I am looking for better name for that class https://github.com/hmcts/snl-events/pull/78/files#diff-7bd548598175fce70af6d2bec5be84b2

Reply:
OurMockMvc->MockMvcPlus, MockMvcExpanded, http://maxgfeller.com/javaclassnamegenerator/